release/run_release_test.sh

```diff
@@ -69,6 +69,23 @@ if [ -z "${NO_CLONE}" ]; then
   echo "Cloning test repo ${RAY_TEST_REPO} branch ${RAY_TEST_BRANCH}"
   git clone --depth 1 -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" "${TMPDIR}"
   pushd "${TMPDIR}/release" || true
+  HEAD_COMMIT=$(git rev-parse HEAD)
```

Comment:
Instead of try to sync later, I wonder if here we can just parse the commit and branch from the installed wheel, and use that information to co a right clone/checkout here?